Point.h  compilation error Windows

I get an error when compiling Kratos with Visual Studio. 

As point.h is not longer a template, typename cannot be used in Visual Studio

C:\Kratos\kratos\geometries/point.h(83): error C2899: typename cannot be used outside a template declaration

@pooyan-dadvand  I guess you can simply delete them right?

--- a/kratos/geometries/point.h
+++ b/kratos/geometries/point.h
@@ -80,9 +80,9 @@ public:
 
     typedef BaseType CoordinatesArrayType;
 
-    typedef typename std::size_t SizeType;
+    typedef std::size_t SizeType;
 
-    typedef typename std::size_t IndexType;
+    typedef std::size_t IndexType;
 
     ///@}
     ///@name Life Cycle

--- a/kratos/spaces/ublas_space.h
+++ b/kratos/spaces/ublas_space.h
@@ -859,7 +859,7 @@ private:
         }
     }
 
-    static void CreatePartition(unsigned int number_of_threads, const int number_of_rows, vector<unsigned int>& partitions)
+    static void CreatePartition(unsigned int number_of_threads, const int number_of_rows, boost::numeric::ublas::vector<unsigned int>& partitions)
     {
         partitions.resize(number_of_threads + 1);
         int partition_size = number_of_rows / number_of_threads;

--- a/kratos/utilities/condition_number_utility.h
+++ b/kratos/utilities/condition_number_utility.h
@@ -65,13 +65,13 @@ public:
     
     typedef UblasSpace<double, Matrix, Vector> LocalSpaceType;
 
-    typedef typename SparseSpaceType::MatrixType SparseMatrixType;
+    typedef SparseSpaceType::MatrixType SparseMatrixType;
 
-    typedef typename SparseSpaceType::VectorType VectorType;
+    typedef SparseSpaceType::VectorType VectorType;
 
-    typedef typename LocalSpaceType::MatrixType DenseMatrixType;
+    typedef LocalSpaceType::MatrixType DenseMatrixType;
 
-    typedef typename LocalSpaceType::VectorType DenseVectorType;
+    typedef LocalSpaceType::VectorType DenseVectorType;
 
     typedef LinearSolver<SparseSpaceType, LocalSpaceType> LinearSolverType;
 

